scipy/stats/tests/test_hypotests.py

```diff
@@ -405,3 +406,105 @@ def test_contingency_table_return(self):
         y = np.arange(10)
         res = stats.somersd(x, y)
         assert_equal(res.table, np.eye(10))
+
+
+class TestBarnardExact(object):
+    """Some tests to show that barnard_exact() works correctly.
+    """
+
+    @pytest.mark.parametrize("input_sample,expected", [
+        ([[43, 40], [10, 39]], (-3.555406779643, 0.000362832367)),
+        ([[100, 2], [1000, 5]], (1.776382925679, 0.135126970878)),
+        ([[2, 7], [8, 2]], (2.518474945157, 0.019210815430)),
+        ([[5, 1], [10, 10]], (-1.449486150679, 0.156277546306)),
+        ([[5, 15], [20, 20]], (1.851640199545, 0.066363501421)),
+        ([[5, 16], [20, 25]], (1.609639949352, 0.116984852192)),
+        ([[10, 5], [10, 1]], (1.449486150679, 0.177536588915)),
+        ([[5, 0], [1, 4]], (-2.581988897472, 0.013671875000)),
+        ([[0, 1], [3, 2]], (1.095445115010, 0.509667991877)),
+        ([[0, 2], [6, 4]], (1.549193338483, 0.197019618792)),
+        ([[2, 7], [8, 2]], (2.518474945157, 0.019210815430)),
+    ])
+    def test_precise(self, input_sample, expected):
+        """The expected values have been generated by R, using a resolution
+        for the nuisance parameter of 1e-6.
```

Comment:
Thanks! Tests against R are great. Can you provide your code for one of these as a comment here? We need to know which R function is used, and I will want to check these myself, so having the code you used for one of them will help.

Reply:
This is quite long, I hope you don't mind. I wrote it in a way there is an easy output for me to copy past in `@pytest.mark.parametrize`. Here it is : 
```
install.packages("Barnard")
library(Barnard)
options(digits=10)

get_data <- function() {
    dp <- 1e-6

    data <- list(
        c(43, 40, 10, 39),
        c(100, 2, 1000, 5),
        c(2, 7, 8, 2),
        c(5, 1, 10, 10),
        c(5, 15, 20, 20),
        c(5, 16, 20, 25),
        c(10, 5, 10, 1),
        c(5, 0, 1, 4),
        c(0, 1, 3, 2),
        c(0, 2, 6, 4),
        c(2, 7, 8, 2)
    )
    for (value in data) {
        sink(file="undesird.txt")
        a = barnard.test(value[1], value[2], value[3], value[4], dp=dp)
        sink()
        # # ([[7, 12], [8, 3]], (1.89433807606, 0.06815343273))
        cat(sprintf("([[%d, %d], [%d, %d]], (%.12f, %.12f)),\n", value[1], value[2], value[3], value[4], a$statistic, a$p.value[2]))
    }

    print("", quote = FALSE)

}

get_data_pooled <- function() {
    dp <- 1e-6

    data <- list(
        c(43, 40, 10, 39),
        c(100, 2, 1000, 5),
        c(2, 7, 8, 2),
        c(5, 1, 10, 10),
        c(5, 15, 20, 20),
        c(5, 16, 20, 25),
        c(10, 5, 10, 1),
        c(5, 0, 1, 4),
        c(0, 1, 3, 2),
        c(0, 2, 6, 4),
        c(2, 7, 8, 2)
    )
    for (value in data) {
        sink(file="undesird.txt")
        a = barnard.test(value[1], value[2], value[3], value[4], dp=dp, pooled=FALSE)
        sink()
        # # ([[7, 12], [8, 3]], (1.89433807606, 0.06815343273))
        cat(sprintf("([[%d, %d], [%d, %d]], (%.12f, %.12f)),\n", value[1], value[2], value[3], value[4], a$statistic, a$p.value[2]))
    }

    print("", quote = FALSE)

}

less_or_greater <- function() {
    dp <- 1e-6

    data <- list(
        c(2, 7, 8, 2),
        c(200, 7, 8, 300),
        c(28, 21, 6, 1957),
        c(190, 800, 200, 900)
    )
    for (value in data) {
        sink(file="undesird.txt")
        a = barnard.test(value[1], value[2], value[3], value[4], dp=dp, )
        sink()
        # # ([[7, 12], [8, 3]], (1.89433807606, 0.06815343273))
        cat(sprintf("([[%d, %d], [%d, %d]], (%.12f, %.12f)),\n", value[1], value[2], value[3], value[4], a$statistic, a$p.value[1]))
    }

    print("", quote = FALSE)

}

print("get_data ss")
get_data()
print("get_data_pooled")
get_data_pooled()
print("less_or_greater")
less_or_greater()
```

Careful, with `dp=1e-6`, the execution time is very long. If you want to see the full output, here it is : 
```
1] "get_data ss"
([[43, 40], [10, 39]], (-3.555406779643, 0.000362832367)),
([[100, 2], [1000, 5]], (1.776382925679, 0.135126970878)),
([[2, 7], [8, 2]], (2.518474945157, 0.019210815430)),
([[5, 1], [10, 10]], (-1.449486150679, 0.156277546306)),
([[5, 15], [20, 20]], (1.851640199545, 0.066363501421)),
([[5, 16], [20, 25]], (1.609639949352, 0.116984852192)),
([[10, 5], [10, 1]], (1.449486150679, 0.177536588915)),
([[5, 0], [1, 4]], (-2.581988897472, 0.013671875000)),
([[0, 1], [3, 2]], (1.095445115010, 0.509667991877)),
([[0, 2], [6, 4]], (1.549193338483, 0.197019618792)),
([[2, 7], [8, 2]], (2.518474945157, 0.019210815430)),
[1] 
[1] "get_data_pooled"
([[43, 40], [10, 39]], (-3.920362887717, 0.000289470662)),
([[100, 2], [1000, 5]], (1.139432816087, 0.950272080594)),
([[2, 7], [8, 2]], (3.079373904042, 0.020172119141)),
([[5, 1], [10, 10]], (-1.622375939458, 0.150599922226)),
([[5, 15], [20, 20]], (1.974771239528, 0.063038448651)),
([[5, 16], [20, 25]], (1.722122973346, 0.133329494287)),
([[10, 5], [10, 1]], (1.765469659009, 0.250566655215)),
([[5, 0], [1, 4]], (-5.477225575052, 0.007812500000)),
([[0, 1], [3, 2]], (1.224744871392, 0.509667991877)),
([[0, 2], [6, 4]], (1.732050807569, 0.197019618792)),
([[2, 7], [8, 2]], (3.079373904042, 0.020172119141)),
[1] 
[1] "less_or_greater"
([[2, 7], [8, 2]], (2.518474945157, 0.009886140845)),
([[200, 7], [8, 300]], (-21.320036698460, 0.000000000000)),
([[28, 21], [6, 1957]], (-30.489638143953, 0.000000000000)),
```

Reply:
> Can you provide your code for one of these as a comment here

Please eliminate all the looping. As a comment in your test, just include the code for one of them like:
```
library(Barnard)
options(digits=10)
barnard.test(value[1], value[2], value[3], value[4], dp=dp, pooled=FALSE)
```
but with values entered in.

Reply:
Ok, I thought you were talking about reviews' comments. I'll do that